src/GraphQL/Execution/Nodes/ExecutionNode.cs

```diff
@@ -125,7 +105,7 @@ public IObjectGraphType GetParentType(ISchema schema)
             if (parentType is IObjectGraphType objectType)
                 return objectType;
 
-            if (parentType is IAbstractGraphType abstractType && Parent.IsResultSet)
+            if (parentType is IAbstractGraphType abstractType)
```

Comment:
I'm not sure exactly when this executes, but after running all the tests, the parent always had its result set.  I believe that is implicit, as the child nodes are not constructed until the parent node is executed (so that if the result is null, for example, no child node is constructed).